manifester/helpers.py

```diff
@@ -171,15 +171,43 @@ def load_inventory_file(file):
             return yaml.load(f, Loader=yaml.FullLoader) or []
 
 
-def update_inventory(inventory_data):
+def _dump_inventory_file(inventory_path, inventory):
+    """Write inventory data to local inventory file."""
+    with inventory_path.open("w") as inventory_file:
+        yaml.dump(inventory, inventory_file, allow_unicode=True)
+
+
+def _update_inventory(inventory_path, inventory, allocation):
+    """Add new allocation, remove and recreate inventory file."""
+    inventory.append(allocation)
+    inventory_path.unlink()
+    inventory_path.touch()
+    _dump_inventory_file(inventory_path, inventory)
```

Comment:
could this be simplified?
```suggestion
def _update_inventory(inventory_path, inventory, allocation):
    """Add new allocation, remove and recreate inventory file."""
    inventory.append(allocation)
    inventory_path.unlink()
    inventory_path.touch()
    yaml.dump(inventory, inventory_path, allow_unicode=True)
```

Reply:
@JacobCallahan Per the PyYAML docs, `yaml.dump` needs an open file in order to write to it directly. Your proposed changed throws an `AttributeError: 'PosixPath' object has no attribute 'write'. Did you mean: 'drive'?` when it tries to dump to the `Path` object. So we need a second line before `yaml.dump` to open the inventory file. I think it's slightly cleaner to move that action into a separate helper function since it gets used in three places, but that part isn't strictly necessary.